spring-integration-mail/src/main/java/org/springframework/integration/mail/dsl/ImapIdleChannelAdapterSpec.java

```diff
@@ -0,0 +1,373 @@
+		assertReceiver();
+		this.receiver.setShouldMarkMessagesAsRead(shouldMarkMessagesAsRead);
+		return this;
+	}
+
+	/**
+	 * Set the name of the flag to use to flag messages when the server does
+	 * not support \Recent but supports user flags;
+	 * default {@value ImapMailReceiver#DEFAULT_SI_USER_FLAG}.
+	 * @param userFlag the flag.
+	 * @return the spec.
+	 * @see ImapMailReceiver#setUserFlag(String)
+	 */
+	public ImapIdleChannelAdapterSpec userFlag(String userFlag) {
+		assertReceiver();
+		this.receiver.setUserFlag(userFlag);
+		return _this();
+	}
+
+	/**
+	 * Set the header mapper; if a header mapper is not provided, the message payload is
+	 * a {@link MimeMessage}, when provided, the headers are mapped and the payload is
+	 * the {@link MimeMessage} content.
+	 * @param headerMapper the header mapper.
+	 * @return the spec.
+	 * @see ImapMailReceiver#setUserFlag(String)
+	 * @see #embeddedPartsAsBytes(boolean)
+	 */
+	public ImapIdleChannelAdapterSpec headerMapper(HeaderMapper<MimeMessage> headerMapper) {
+		assertReceiver();
+		this.receiver.setHeaderMapper(headerMapper);
+		return _this();
+	}
+
+	/**
+	 * When a header mapper is provided determine whether an embedded {@link Part} (e.g
+	 * {@link Message} or {@link javax.mail.Multipart} content is rendered as a byte[] in
+	 * the payload. Otherwise, leave as a {@link Part}. These objects are not suitable for
+	 * downstream serialization. Default: true.
+	 * <p>
+	 * This has no effect if there is no header mapper, in that case the payload is the
+	 * {@link MimeMessage}.
+	 * @param embeddedPartsAsBytes the embeddedPartsAsBytes to set.
+	 * @return the spec.
+	 * @see #headerMapper(HeaderMapper)
+	 */
+	public ImapIdleChannelAdapterSpec embeddedPartsAsBytes(boolean embeddedPartsAsBytes) {
+		assertReceiver();
+		this.receiver.setEmbeddedPartsAsBytes(embeddedPartsAsBytes);
+		return _this();
+	}
+
+
+	/**
+	 * Configure a {@link TransactionSynchronizationFactory}. Usually used to synchronize
+	 * message deletion with some external transaction manager.
+	 * @param transactionSynchronizationFactory the transactionSynchronizationFactory.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactionSynchronizationFactory(
+			TransactionSynchronizationFactory transactionSynchronizationFactory) {
+		this.target.setTransactionSynchronizationFactory(transactionSynchronizationFactory);
+		return this;
+	}
+
+	/**
+	 * Configure a chain of {@link Advice} objects for message delivery.
+	 * @param adviceChain the advice chain.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec adviceChain(Advice... adviceChain) {
+		this.adviceChain.addAll(Arrays.asList(adviceChain));
+		return this;
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@code pollingTask}.
```

Comment:
I know there is nothing about `pollingTask` in the `ImapIdleChannelAdapter`, that's why I decided to take a look to JavaDocs where you have borrowed this :smile: